lib/cloudregister/registerutils.py

```diff
@@ -1442,6 +1442,31 @@ def write_framework_identifier(cfg):
         framework_file.write(json.dumps(identifier))
 
 
+# ----------------------------------------------------------------------------
+def has_ipv4_access():
+    """Check if we have IPv4 network configuration"""
+    return has_network_access_by_ipversion('8.8.8.8')
+
+
+# ----------------------------------------------------------------------------
+def has_ipv6_access():
+    """Check if we have IPv6 network configuration"""
+    return has_network_access_by_ipversion('2001:4860:4860::8888')
+
+
+# ----------------------------------------------------------------------------
+def has_network_access_by_ipversion(server_ip):
+    """Check if we can connect to the given server"""
+    try:
+        rmt_connection = socket.create_connection((server_ip, 443), timeout=2)
```

Comment:
s/rmt_connection/connection/ we are getting any IP address as such the variable name should be generic and not imply a connection to a specific system.

Reply:
Updated